Python/thread_pthread.h

```diff
@@ -399,6 +401,8 @@ PyThread_get_thread_native_id(void)
 #elif defined(__DragonFly__)
     lwpid_t native_id;
     native_id = lwp_gettid();
+#elif defined(__sun__)
+    unsigned long native_id = (unsigned long)getpid() << 32 | thr_self();
```

Comment:
Why is this? How does it work on 32-bit Solaris?

Reply:
Well, I see now your explanation, but do 32-bit Solaris or Illumos exist?

Reply:
Illumos doesn't have a 32-bit kernel build since 2018:
* https://www.illumos.org/issues/8685
* http://www.tribblix.org/32bit.html

It seems there's no support for 32bit hardware on Solaris 11, but I couldn't find the release which removed that:
* https://en.wikipedia.org/wiki/Oracle_Solaris

Solaris 11.4 includes Python 3.4 in its package repository, and it is 64bits.
~~~
$ uname -a
SunOS solaris 5.11 11.4.0.15.0 i86pc i386 i86pc
$ file `which python3`  
/usr/bin/python3:       ELF 64-bit LSB executable AMD64 Version 1 [SSE], dynamically linked, not stripped
~~~

Reply:
That makes sense. Addressed at https://github.com/python/cpython/pull/137927/commits/8c1140916fbb1e07389d5ee7d06f15b24f7bbb9a

Reply:
You need to add the conditions also in other places.

Reply:
Updated at: https://github.com/python/cpython/pull/137927/commits/e2f9c41905d629206bd3366fbaa25d86d6663abc